readme.md

```diff
@@ -0,0 +1,330 @@
+| keycloak.public.client.id | Public Client ID registered in the Keycloak that is used in the plugin process. | gui | string  | BS_KEYCLOAK_PUBLIC_CLIENT_ID
+| keycloak.code.challenge.method | How is the code_challenge is hashed. S256=SHA-256 | S256 | string | BS_CODE_CHALLENGE_METHOD
+| keycloak.secure |  If you want to verify the SSL Certs | false | boolean  | BS_KEYCLOAK_SECURE
+| keycloak.ssl.ca | File path to CAs. | none | path  | BS_KEYCLOAK_CLIENT_TLS_CA
+| keycloak.ssl.key | File path to key certificate. | none | path  | BS_KEYCLOAK_SSL_KEY
+| keycloak.ssl.cert | File path to  certificate. | none | path  | BS_KEYCLOAK_SSL_CERT
+| keycloak.ssl.request.cert |  Whether to authenticate the remote peer by requesting a certificate. Clients always request a server certificate. | none | boolean | BS_KEYCLOAK_SSL_REQUEST_CERT
+| keycloak.ssl.reject.unauthorized |   If true, the server certificate is verified against the list of supplied CAs.  | none | boolean |  BS_KEYCLOAK_SSL_REJECT_UNAUTHORIZED
+
+#### Session Configurations
+
+| Key | Purpose | Default Value | Valid Values | Environment variable
+| --- | ------- | ------------- | ------------ | --------------------
+| session.secret | This is the secret used to sign the session ID cookie.  **NOTE: Should be unique for each environment** | mustbechanged | string  | BS_SESSION_SECRET
+| session.cookie.name | The name of the session ID cookie to set in the response (and read from in the request) | dojot-backstage-cookie | string  | BS_SESSION_COOKIE_NAME
+| session.cookie.https | True is a recommended option. However, it requires an https-enabled website, i.e., HTTPS is necessary for secure cookies. If secure is set, and you access your site over HTTP, the cookie will not be set. If you have your node.js behind a proxy and are using secure: true, you need to set "trust proxy" in express
+ | true | boolean  | BS_SESSION_COOKIE_HTTPS
+| session.cookie.path | Specifies the value for the Path Set-Cookie. By default, this is set to '/', which is the root path of the domain. | / | string  | BS_SESSION_COOKIE_PATH
+| session.proxy  | Trust the reverse proxy when setting secure cookies (via the "X-Forwarded-Proto" header). |true | boolean  | BS_SESSION_PROXY
+| session.domain | The domain where the cookie will be used.  Specifies the value for the Domain Set-Cookie attribute.  |localhost | string  | BS_SESSION_DOMAIN
+| session.redis.max.login.return.time.sec |  Maximum time before the session expires when starting a login and being returned to the `auth/return` endpont.   | 120 | number (seconds) | BS_SESSION_REDIS_MAX_LOGIN_RETURN_TIME_SEC
+| session.redis.max.life.time.sec |  Maximum lifetime of a session. |86400 | number (seconds) | BS_SESSION_REDIS_MAX_LIFE_TIME_SEC
+| session.redis.max.idle.time.sec | Maximum time that a session can be idle. |1800 | number (seconds) | BS_SESSION_REDIS_MAX_IDLE_TIME_SEC
```

Comment:
The configuration key is duplicated

Reply:
Regarding the first configuration, `Maximum lifetime of a session` It would be nice to mention that this value needs to be synchronized with the user session time defined in the keycloak or set a value large enough to accommodate the keycloak session time.

Regarding the second configuration, `Maximum time that a session can be idle` I think it is better to leave this configuration (sliding window) only in the Keycloak, that is, as long as it is possible to renew the access token, then the session will be active. Let the keycloak control session expiration due to inactivity.

Reply:
I will document the relationship between the `SSO Session Idle` and `SSO Session Max` to `session.redis.max.life.time.sec ` and `session.redis.max.idle.time.sec`